app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+class User < ActiveRecord::Base
+  has_and_belongs_to_many :twitter_users
+  has_and_belongs_to_many :vimeo_users
+
+  validates :name, presence: true
+  validates :uid, presence:true, uniqueness: true
+  validates :provider, presence: true
+
+  def self.find_or_create_from_omniauth(auth_hash)
+    user = self.find_by(uid: auth_hash["uid"], provider: auth_hash["provider"])
+    if !user.nil?
+      #user found, return user and continue on with your life
+      return user
+    else
+      # create a new user.
+      user = User.new
+      user.uid = auth_hash["uid"]
+      user.provider = auth_hash["provider"]
+      user.name = auth_hash["info"]["name"]
+      user.image = auth_hash["info"]["image"]
+      if user.save
+        return user
+      else
+        return nil
```

Comment:
if you kept the method more like this, you should raise an error here instead of returning nil to illuminate the fact that the user was unable to be created :bulb: